primo/data_parser/metric_data.py

```diff
@@ -423,13 +443,29 @@ def register_new_metric(
         name : str
             Metric name, must be a valid python variable name
 
-        weight : int
-            weight associated with the metric
-
-        full_name : str
+        full_name : str, default = None
             Elaborate name of the metric
+
+        is_binary_type : bool, default = False
+            Is the admissible value for this metric of boolean type?
+
+        has_inverse_priority : bool, default = False
+            Is the priority associated with this metric inversely
+            proportional to the value/magnitude of the metric?
+
+        data_col_name : str, default = None
+            Name of the column containing the data corresponding to
+            this metric
+
+        fill_missing_value : Any, default = None
+            Default value for missing values corresponding to this metric
         """
-        setattr(self, name, Metric(name=name, weight=weight, full_name=full_name))
+        setattr(self, name, Metric(name=name, weight=0, full_name=full_name))
```

Comment:
Curious why weight is no longer an argument? Are we forcing an explicit set_weight?

Reply:
Yeah. The preferred workflow to avoid confusion (and the possibility of making errors) is:
```python
im_metrics = ImpactMetrics()

im_metrics.register_new_metric(...)
im_metrics.register_new_submetrics(...)
im_metrics.delete_metric(...)

im_metrics.set_weight({....})
```
@DevKakkad32 was trying

Reply:
I see. I ll allow it lol